include/eve/arch/riscv/tags.hpp

```diff
@@ -0,0 +1,118 @@
+
+  template<typename Type>
+  static constexpr std::size_t fundamental_cardinal = Size / 8 / sizeof(Type);
+
+  template<typename Type>
+  static constexpr std::size_t expected_cardinal = Size / 8 / sizeof(Type) * MaxLmul;
+
+  template<typename Type> static consteval int get_min_frac_lmul()
+  {
+    auto type_size = sizeof(Type);
+    if( type_size == 1 ) return 8;
+    if( type_size == 2 ) return 4;
+    if( type_size == 4 ) return 2;
+    // for bigger types we can not use frac lmul.
+    // Return bigger than size of vector register value.
+    return Size + 1;
+  }
+
+  template<typename Type> static consteval int get_max_frac_lmul()
+  {
+    auto type_size = sizeof(Type);
+    if( type_size == 1 ) return 2;
+    if( type_size == 2 ) return 2;
+    if( type_size == 4 ) return 2;
+    // for bigger types we can not use frac lmul.
+    // Return bigger than size of vector register value.
+    return Size + 1;
+  }
+
+  // return natural lmul if > 0, frac otherwise
+  template<typename Type> static consteval int getLMUL(size_t cardinal)
+  {
+    auto binary_size   = cardinal * sizeof(Type) * 8;
+    auto min_frac_lmul = get_min_frac_lmul<Type>();
+    auto max_frac_lmul = get_max_frac_lmul<Type>();
+
+    auto frac_lmul_border = Size / max_frac_lmul;
+    if( binary_size <= frac_lmul_border )
+    {
+      auto frac_lmul = Size / binary_size;
+      frac_lmul      = std::bit_floor(frac_lmul);
+      if( frac_lmul > min_frac_lmul ) frac_lmul = min_frac_lmul;
+      return -frac_lmul;
+    }
+    auto lmul = binary_size / Size;
+    if( binary_size % Size ) ++lmul;
+    return std::bit_ceil(lmul);
+  }
+
+  template<arithmetic_scalar_value T> using m1_fixed = fixed<Size / 8 / sizeof(T)>;
+};
+
+#ifdef __riscv_v_fixed_vlen
+struct riscv_rvv_dyn_ : rvv_abi_<__riscv_v_fixed_vlen>
+{};
+#else
+struct riscv_rvv_dyn_ : rvv_abi_<1>
+{};
+#endif
+
+//================================================================================================
+// Dispatching tag for ARM SIMD implementation
+//================================================================================================
+struct rvv_ : simd_api<simd_, spy::rvv_>
+{
+  using is_rvv = void;
+};
+struct rvv_api_ : simd_api<rvv_, spy::rvv_>
+{
+  using is_rvv = void;
+};
+
+//================================================================================================
+// SVE extensions tag objects
+//================================================================================================
+inline constexpr rvv_     rvv     = {};
+inline constexpr rvv_api_ rvv_api = {};
+
+//================================================================================================
+// RISCV SVE ABI concept
+//================================================================================================
+template<typename T>
+concept rvv_abi = detail::is_one_of<T>(detail::types<riscv_rvv_dyn_> {});
```

Comment:
why dyn

Reply:
Well, I wanted to note that we should use this template for any VLEN. I could rename, if you want.